phoenix-core/src/main/java/org/apache/phoenix/coprocessor/tasks/TransformMonitorTask.java

```diff
@@ -0,0 +1,215 @@
+                LOGGER.error(str);
+                return new TaskRegionObserver.TaskResult(TaskRegionObserver.TaskResultCode.FAIL, str);
+            }
+
+            // Update task status to RETRY so that it is retried
+            Task.addTask(new SystemTaskParams.SystemTaskParamsBuilder()
+                    .setConn(conn)
+                    .setTaskType(taskRecord.getTaskType())
+                    .setTenantId(taskRecord.getTenantId())
+                    .setSchemaName(taskRecord.getSchemaName())
+                    .setTableName(taskRecord.getTableName())
+                    .setTaskStatus(PTable.TaskStatus.RETRY.toString())
+                    .setData(taskRecord.getData())
+                    .setPriority(taskRecord.getPriority())
+                    .setStartTs(taskRecord.getTimeStamp())
+                    .setEndTs(null)
+                    .setAccessCheckEnabled(true)
+                    .build());
+            return null;
+        }
+        catch (Throwable t) {
+            LOGGER.warn("Exception while running transform monitor task. " +
+                    "It will be retried in the next system task table scan : " +
+                    taskRecord.getSchemaName() + "." + taskRecord.getTableName() +
+                    " with tenant id " + (taskRecord.getTenantId() == null ? " IS NULL" : taskRecord.getTenantId()) +
+                    " and data " + taskRecord.getData(), t);
+            return new TaskRegionObserver.TaskResult(TaskRegionObserver.TaskResultCode.FAIL, t.toString());
+        }
+    }
+
+    public static void addTransformMonitorTask(PhoenixConnection connection, Configuration configuration, SystemTransformRecord systemTransformRecord,
+                                               PTable.TaskStatus taskStatus, Timestamp startTimestamp, Timestamp endTimestamp) throws IOException {
+        boolean transformMonitorEnabled = configuration.getBoolean(TRANSFORM_MONITOR_ENABLED, DEFAULT_TRANSFORM_MONITOR_ENABLED);
+        if (!transformMonitorEnabled) {
+            LOGGER.warn("TransformMonitor is not enabled. Monitoring/retrying TransformTool and doing cutover will not be done automatically");
+            return;
+        }
+
+        Task.addTask(new SystemTaskParams.SystemTaskParamsBuilder()
+                .setConn(connection)
+                .setTaskType(PTable.TaskType.TRANSFORM_MONITOR)
+                .setTenantId(systemTransformRecord.getTenantId())
+                .setSchemaName(systemTransformRecord.getSchemaName())
+                .setTableName(systemTransformRecord.getLogicalTableName())
+                .setTaskStatus(taskStatus.toString())
+                .setStartTs(startTimestamp)
+                .setEndTs(endTimestamp)
+                .setAccessCheckEnabled(true)
+                .build());
+    }
+
+    @Override
+    public TaskRegionObserver.TaskResult checkCurrentResult(Task.TaskRecord taskRecord)
+            throws Exception {
+        return null;
```

Comment:
why is this returning null? Think we need a comment at least please.

Reply:
Will add a comment but here is the explanation. IndexRebuildTask uses this function to check if MR job succeeded or not. For this task, we don't need it. The MR job itself updates the task status.